### ISSUE TITLE
Filtering a single validator by name takes over 30s on Kusama due to style recalculation

### ISSUE THREAD

**Issue by DemiMarie** [polkadot-js/apps#3458]
If I go to https://polkadot.js.org/apps/#/staking and filter out all but one validator using the search box, the UI takes over 30s to stabilize.  I looked at the performance waterfall using Firefox’s development tools, which indicated that a massive amount of time is being spent on restyling.

Batching style updates should help this.

**Comment by jacogr** [comment]
Known issue and will just get worse at this point.

There is no "batching style updates" functionality in React. FF is showing the effect, not the actual root cause. Sadly it can just interpret what it knows about and then leads down the wrong path. Underlying this is a state update issue since is not a traditional list, each of the items has to filter themselves and decide if they are visible or not. (Which means on any change there are state updates for each item... and only then rendering)

It has an effect everywhere, anywhere where you can filter, atm the only solution is actually not to filter by name at all. There is ongoing work to keep a cache of names globally (as retrieved), which can be used to make the lists behave as actual lists (and would mean 1 state update as opposed to X.) But it is still some ways off.

TL;DR Not a UI display issue.

**Comment by jacogr** [comment]
Closing. Ongoing.
